api/src/deployment.rs

```diff
@@ -488,21 +484,15 @@ impl DeploymentSystem {
                 // library when the runtime gets around to it.
 
                 let mut lock = deployment.state.write().await;
-                if let DeploymentState::Deployed(DeployedState {
-                    so,
-                    handle,
-                    shutdown_handle,
-                    ..
-                }) = lock.take()
-                {
-                    if let Some(f) = *shutdown_handle {
-                        // TODO: ideally it would receive feedback regarding shutdown
-                        // finalization, and only then we'd move to aborting the thread
-                        // TODO: error handling
-                        f.send(true).ok();
-                    }
-                    // QUESTION: why abort thread instead of stopping the runtime?
-                    handle.abort();
+                if let DeploymentState::Deployed(DeployedState { so, service, .. }) = lock.take() {
+                    // TODO: error handling
+                    let mut service = service.lock().unwrap();
```

Comment:
The api is getting frozen because we're not able to acquire lock here. That happens because `bind()` is blocking.